installscripts/jazz-terraform-unix-noinstances/chefProvisioner.tf

```diff
@@ -104,27 +104,130 @@ resource "null_resource" "configureJenkinsInstance" {
   }
 }
+
+# For scenario2 (Jenkins ALB)
+# For new vpc
+resource "null_resource" "configureCliJenkinsbb_dockerized" {
+  count = "${var.autovpc * var.scmbb * var.dockerizedJenkins}"
+  depends_on = ["null_resource.preJenkinsConfiguration"]
+  #Jenkins Cli process
+  provisioner "local-exec" {
+    command = "bash ${var.configureJenkinsCE_cmd} ${aws_lb.alb_ecs.dns_name} ${var.cognito_pool_username} ${var.dockerizedJenkins} ${lookup(var.scmmap, "scm_elb")} ${lookup(var.scmmap, "scm_username")} ${lookup(var.scmmap, "scm_passwd")} ${lookup(var.scmmap, "scm_passwd")} ${lookup(var.jenkinsservermap, "jenkinspasswd")} ${lookup(var.scmmap, "scm_type")} ${lookup(var.codeqmap, "sonar_username")} ${lookup(var.codeqmap, "sonar_passwd")} ${aws_iam_access_key.operational_key.id} ${aws_iam_access_key.operational_key.secret} ${var.cognito_pool_password} ${lookup(var.jenkinsservermap, "jenkinsuser")}"
+  }
+}
+# For existing vpc
+resource "null_resource" "configureCliJenkinsbb_dockerized_existing" {
+  count = "${(var.dockerizedJenkins - var.autovpc) * var.scmbb * var.dockerizedJenkins}"
+  depends_on = ["null_resource.preJenkinsConfiguration"]
+  #Jenkins Cli process
+  provisioner "local-exec" {
+    command = "bash ${var.configureJenkinsCE_cmd} ${aws_lb.alb_ecs_existing.dns_name} ${var.cognito_pool_username} ${var.dockerizedJenkins} ${lookup(var.scmmap, "scm_elb")} ${lookup(var.scmmap, "scm_username")} ${lookup(var.scmmap, "scm_passwd")} ${lookup(var.scmmap, "scm_passwd")} ${lookup(var.jenkinsservermap, "jenkinspasswd")} ${lookup(var.scmmap, "scm_type")} ${lookup(var.codeqmap, "sonar_username")} ${lookup(var.codeqmap, "sonar_passwd")} ${aws_iam_access_key.operational_key.id} ${aws_iam_access_key.operational_key.secret} ${var.cognito_pool_password} ${lookup(var.jenkinsservermap, "jenkinsuser")}"
+  }
+}
+
+# For scenario3 (Jenkins and Gitlab ALB)
+# For new vpc
+resource "null_resource" "configureCliJenkins_dockerized" {
+  count = "${var.autovpc * var.scmgitlab * var.dockerizedJenkins}"
+  depends_on = ["null_resource.preJenkinsConfiguration"]
+  #Jenkins Cli process
+  provisioner "local-exec" {
+    command = "bash ${var.configureJenkinsCE_cmd} ${aws_lb.alb_ecs.dns_name} ${var.cognito_pool_username} ${var.dockerizedJenkins} ${aws_lb.alb_ecs_gitlab.dns_name} ${lookup(var.scmmap, "scm_username")} ${lookup(var.scmmap, "scm_passwd")} ${data.external.gitlabcontainer.result.token} ${lookup(var.jenkinsservermap, "jenkinspasswd")} ${lookup(var.scmmap, "scm_type")} ${lookup(var.codeqmap, "sonar_username")} ${lookup(var.codeqmap, "sonar_passwd")} ${aws_iam_access_key.operational_key.id} ${aws_iam_access_key.operational_key.secret} ${var.cognito_pool_password} ${lookup(var.jenkinsservermap, "jenkinsuser")}"
```

Comment:
Again, all 3 of these blocks call `var.configureJenkinsCE_cmd` with 99% identical arguments. Create one block, and use `locals` to conditionally set the value of the 1 or 2 variables that do actually differ.

Reply:
Removed duplicates.